### ISSUE TITLE
UnusedPrivateClass false positive with private annotation

### ISSUE THREAD

**Issue by cordis-dev** [arturbosch/detekt#2093]
## Expected Behavior
No UnusedPrivateClass

## Observed Behavior
UnusedPrivateClass - `Test`

## Steps to Reproduce
```kotlin
private annotation class Test

@Test class Custom {}
```

Private data annotation class is used for `Custom` class, but it's reported as unused.

## Your Environment
* Version of detekt used: `1.1.1`
* Operating System and version: Win 7 SP1
